src/constructors/regular.c

```diff
@@ -139,6 +139,155 @@ igraph_error_t igraph_star(igraph_t *graph, igraph_integer_t n, igraph_star_mode
+ *
+ */
+
+igraph_error_t igraph_wheel(igraph_t *graph, igraph_integer_t n, igraph_wheel_mode_t mode,
+                igraph_integer_t center) {
+
+    igraph_star_mode_t star_mode;
+    igraph_vector_int_t rim_edges = IGRAPH_VECTOR_NULL;
+    igraph_integer_t i;
+
+    /* create a star and also make use of its existing input parameter checking, */
+    /* error if "Invalid number of vertices" and "Invalid center vertex".        */
+
+    switch (mode)
+    {
+        case IGRAPH_WHEEL_OUT:
+            star_mode = IGRAPH_STAR_OUT;
+            break;
+        case IGRAPH_WHEEL_IN:
+            star_mode = IGRAPH_STAR_IN;
+            break;
+        case IGRAPH_WHEEL_MUTUAL:
+            star_mode = IGRAPH_STAR_MUTUAL;
+            break;
+        case IGRAPH_WHEEL_UNDIRECTED:
+            star_mode = IGRAPH_STAR_UNDIRECTED;
+            break;
+        default:
+            IGRAPH_ERROR("Invalid wheel graph mode.", IGRAPH_EINVMODE);
+    }
+
+    IGRAPH_CHECK(igraph_star(graph, n, star_mode, center));
+
+    /* If n <= 1, wheel graph is identical with star graph, */
+    /* no further processing is needed.                     */
+
+    if (n <= 1) {
+        return IGRAPH_SUCCESS;
+    }
+
+    /* Register the star for deallocation in case of error flow before */
+    /* the entire wheel is successfully created.                       */
+
+    IGRAPH_FINALLY(igraph_destroy, graph);
+
+    /* Add edges to the rim. As the rim (or cycle) has n - 1 vertices, */
+    /* it will have n - 1 edges. However, for MUTUAL mode, number of   */
+    /* edges will be double.                                           */
+
+    if (mode == IGRAPH_WHEEL_MUTUAL) {
+        IGRAPH_VECTOR_INT_INIT_FINALLY(&rim_edges, 4 * (n-1));
+    } else {
+        IGRAPH_VECTOR_INT_INIT_FINALLY(&rim_edges, 2 * (n-1));
+    }
+
+    /* Assign first n-1 edges (not MUTUAL). */
+
+    for (i = 0; i < n-2; i++) {
+        if ( i < center ) {
+            VECTOR(rim_edges)[2 * i] = i;
+            if ( i + 1 < center ) {
+                VECTOR(rim_edges)[2 * i + 1] = i + 1;
+            } else {
+                VECTOR(rim_edges)[2 * i + 1] = i + 2;
+            }
+        } else {
+            VECTOR(rim_edges)[2 * i] = i + 1;
+            VECTOR(rim_edges)[2 * i + 1] = i + 2;
+        }
+    }
+
+    /* Assign the last edge (not MUTUAL). */
+
+    if ( n - 2 < center ) {
+        VECTOR(rim_edges)[2 * n - 4] = n - 2;
+    } else {
+        VECTOR(rim_edges)[2 * n - 4] = n - 1;
+    }
+    if ( center > 0 ) {
+        VECTOR(rim_edges)[2 * n - 3] = 0;
+    } else {
+        VECTOR(rim_edges)[2 * n - 3] = 1;
+    }
+
+    /* If MUTUAL, add reverse-direction edges. */
+
+    if (mode == IGRAPH_WHEEL_MUTUAL) {
+        for (i=0; i < 2 * (n-1); i++) {
+            VECTOR(rim_edges)[4 * (n-1) - 1 - i] = VECTOR(rim_edges)[i];
+        }
+    }
+
+    /* Combine the rim into the star to make it a wheel graph */
+
+    IGRAPH_CHECK(igraph_add_edges(graph, &rim_edges, 0));
```

Comment:
For better readability:

```suggestion
    IGRAPH_CHECK(igraph_add_edges(graph, &rim_edges, NULL));
```

Reply:
Done.